### PR TITLE
:package: feat(Java): Update Java version to 11

### LINKED ISSUE #31: NoClassDefFound on newer Jenkins versions
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
Jenkins: 2.378
OS: Linux - 5.5.10-200.fc31.x86_64
---
ace-editor:1.1
additional-metrics:45.vc0cdf94c7d5b_
ansible:1.1
ansible-tower:0.16.0
ansicolor:1.0.2
ant:481.v7b_09e538fcca
antisamy-markup-formatter:155.v795fb_8702324
anything-goes-formatter:1.0
apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61
authentication-tokens:1.4
aws-credentials:191.vcb_f183ce58b_9
aws-java-sdk-ec2:1.12.287-357.vf82d85a_6eefd
aws-java-sdk-minimal:1.12.287-357.vf82d85a_6eefd
basic-branch-build-strategies:71.vc1421f89888e
blueocean:1.25.8
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.25.8
blueocean-commons:1.25.8
blueocean-config:1.25.8
blueocean-core-js:1.25.8
blueocean-dashboard:1.25.8
blueocean-display-url:2.4.1
blueocean-events:1.25.8
blueocean-git-pipeline:1.25.8
blueocean-github-pipeline:1.25.8
blueocean-i18n:1.25.8
blueocean-jira:1.25.8
blueocean-jwt:1.25.8
blueocean-personalization:1.25.8
blueocean-pipeline-api-impl:1.25.8
blueocean-pipeline-editor:1.25.8
blueocean-pipeline-scm-api:1.25.8
blueocean-rest:1.25.8
blueocean-rest-impl:1.25.8
blueocean-web:1.25.8
bootstrap4-api:4.6.0-5
bootstrap5-api:5.2.1-3
bouncycastle-api:2.26
branch-api:2.1051.v9985666b_f6cc
build-history-metrics-plugin:97.vdcf3aa48af8d
build-monitor-plugin:1.13+build.202205140447
build-name-setter:2.2.0
build-timeout:1.25
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
changelog-history:1.7
changes-since-last-success:0.6
checks-api:1.8.0
cloudbees-bitbucket-branch-source:791.vb_eea_a_476405b
cloudbees-folder:6.800.v71307ca_b_986b
cobertura:1.17
code-coverage-api:3.4.0
command-launcher:90.v669d7ccb_7c31
commons-lang3-api:3.12.0-36.vd97de6465d5b_
commons-text-api:1.10.0-27.vb_fa_3896786a_7
copyartifact:1.48
credentials:1214.v1de940103927
credentials-binding:523.vd859a_4b_122e6
dark-theme:262.v0202a_4c8fb_6a
data-tables-api:1.12.1-4
display-url-api:2.3.6
docker-commons:1.21
docker-workflow:528.v7c193a_0b_e67c
durable-task:503.v57154d18d478
echarts-api:5.4.0-1
email-ext:2.92
embeddable-build-status:304.vdcf48d6b_d2eb
extended-choice-parameter:356.va_90a_94ca_62ec
external-monitor-job:203.v683c09d993b_9
favorite:2.4.1
file-parameters:205.vf6ce13b_e5dee
font-awesome-api:6.2.1-1
forensics-api:1.16.0
git:4.14.1
git-changelog:3.25
git-client:3.13.0
git-server:99.va_0826a_b_cdfa_d
github:1.36.0
github-api:1.303-400.v35c2d8258028
github-autostatus:3.6.2
github-branch-source:1696.v3a_7603564d04
github-oauth:0.39
global-build-stats:244.v27c8a_2e50a_34
google-chat-notification:1.4
gradle:2.1.1
handlebars:3.0.8
handy-uri-templates-2-api:2.1.8-22.v77d5b_75e6953
htmlpublisher:1.31
influxdb:3.4
instance-identity:116.vf8f487400980
ionicons-api:31.v4757b_6987003
jackson2-api:2.14.1-313.v504cdd45c18b
jacoco:3.3.2
jakarta-activation-api:2.0.1-2
jakarta-mail-api:2.0.1-2
javadoc:226.v71211feb_e7e9
javax-activation-api:1.2.0-5
javax-mail-api:1.6.2-8
jaxb:2.3.7-1
jdk-tool:63.v62d2fd4b_4793
jenkins-design-language:1.25.8
jersey2-api:2.37-1
jira:3.8
jira-steps:2.0.141.vd0c6e6dc83f0
jjwt-api:0.11.5-77.v646c772fddb_0
jnr-posix-api:3.1.16-1
jquery:1.12.4-1
jquery-detached:1.2.1
jquery3-api:3.6.1-2
jsch:0.1.55.61.va_e9ee26616e7
junit:1166.va_436e268e972
kubernetes:3734.v562b_b_a_627ea_c
kubernetes-client-api:5.12.2-193.v26a_6078f65a_9
kubernetes-credentials:0.9.0
last-changes:2.7.11
ldap:2.12
list-git-branches-parameter:0.0.11
lockable-resources:1069.v726298f53f8c
mailer:438.v02c7f0a_12fa_4
mapdb-api:1.0.9-28.vf251ce40855d
mask-passwords:3.3
matrix-auth:3.1.5
matrix-project:785.v06b_7f47b_c631
mercurial:1260.vdfb_723cdcc81
metrics:4.2.10-405.v60a_9cc74e923
mina-sshd-api-common:2.9.2-50.va_0e1f42659a_a
mina-sshd-api-core:2.9.2-50.va_0e1f42659a_a
momentjs:1.1.1
monitoring:1.91.0
okhttp-api:4.9.3-108.v0feda04578cf
openJDK-native-plugin:1.5
pam-auth:1.10
periodicbackup:1.9
pipeline-build-step:2.18
pipeline-github:2.8-138.d766e30bb08b
pipeline-github-lib:38.v445716ea_edda_
pipeline-graph-analysis:195.v5812d95a_a_2f9
pipeline-groovy-lib:621.vb_44ce045b_582
pipeline-input-step:466.v6d0a_5df34f81
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2118.v31fd5b_9944b_5
pipeline-model-definition:2.2118.v31fd5b_9944b_5
pipeline-model-extensions:2.2118.v31fd5b_9944b_5
pipeline-rest-api:2.28
pipeline-stage-step:296.v5f6908f017a_5
pipeline-stage-tags-metadata:2.2118.v31fd5b_9944b_5
pipeline-stage-view:2.28
pipeline-utility-steps:2.14.0
plain-credentials:139.ved2b_9cf7587b
plot:2.1.11
plugin-util-api:2.20.0
popper-api:1.16.1-3
popper2-api:2.11.6-2
prism-api:1.29.0-1
pubsub-light:1.17
pyenv-pipeline:2.1.2
python:1.3
resource-disposer:0.20
role-strategy:569.v7476f8e4fe29
rpmsign-plugin:0.5.1
scm-api:621.vda_a_b_055e58f7
script-security:1218.v39ca_7f7ed0a_c
snakeyaml-api:1.33-90.v80dcb_3814d35
solarized-theme:0.1
sse-gateway:1.26
ssh:2.6.1
ssh-agent:295.v9ca_a_1c7cc3a_a_
ssh-credentials:305.v8f4381501156
ssh-slaves:2.854.v7fd446b_337c9
ssh-steps:2.0.39.v831c5e6468b_c
sshd:3.270.vb_a_e71e64c287
structs:324.va_f5d6774f3a_d
subversion:2.16.0
theme-manager:1.5
token-macro:321.vd7cc1f2a_52c8
trilead-api:2.84.v72119de229b_7
uno-choice:2.6.4
variant:59.vf075fe829ccb
windows-slaves:1.8.1
workflow-aggregator:590.v6a_d052e5a_a_b_5
workflow-api:1200.v8005c684b_a_c6
workflow-basic-steps:994.vd57e3ca_46d24
workflow-cps:3536.vb_8a_6628079d5
workflow-durable-task-step:1217.v38306d8fa_b_5c
workflow-job:1254.v3f64639b_11dd
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:839.v35e2736cfd5c
ws-cleanup:0.43 
 ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Fedora 35

### Reproduction steps

Update to Jenkins version 2.379 or newer and try to use the GChat plugin to send a message.

### Expected Results

The message is sent.

### Actual Results

You get a NoClassDefFoundError 
image

### Anything else?

I suspect this is because Jenkins removed support for Commons HttpClient 3.x library for use by plugins.  (see release notes https://www.jenkins.io/changelog/)

### PR DESCRIPTION
**Pull request by regazzoj**
- Use of HttpClient from Java instead of Apache

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Resolves #31

### CODE REVIEW

**Comment by regazzoj** [comment]
@jenkinsadmin Hello I did some changes in Jenkinsfile to use an newer JDK than 8. But it doesn't take effect because I don't have write access.
Can you help me ?

**Review by chiragnaik — APPROVED** [review]

**Review by chiragnaik — CHANGES_REQUESTED** [review]
Fix the failing test and conflicts

**Comment by regazzoj** [comment]
@chiragnaik Let me know if you think some changes could be made.

**Comment by gpflaum** [comment]
The build is failing with this error:
```[2023-03-29T08:44:18.252Z] Caused by: java.lang.UnsupportedClassVersionError: org/jenkinsci/maven/plugins/hpi/ValidateMojo has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0```

Build log shows it is building with Java 8:
```[2023-03-29T08:43:48.131Z] Java version: 1.8.0_362, vendor: Temurin, runtime: C:\tools\jdk-8\jre```

[This blog](https://www.jenkins.io/blog/2022/12/14/require-java-11/#:~:text=If%20you%20neglect%20to%20update%20the%20Jenkinsfile) says that error will happen if Jenkinsfile isn't updated to remove Java 8 configurations. You've done that, and I don't know what else needs to change.

**Comment by regazzoj** [comment]
> The build is failing with this error: `[2023-03-29T08:44:18.252Z] Caused by: java.lang.UnsupportedClassVersionError: org/jenkinsci/maven/plugins/hpi/ValidateMojo has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0`
> 
> Build log shows it is building with Java 8: `[2023-03-29T08:43:48.131Z] Java version: 1.8.0_362, vendor: Temurin, runtime: C:\tools\jdk-8\jre`
> 
> [This blog](https://www.jenkins.io/blog/2022/12/14/require-java-11/#:~:text=If%20you%20neglect%20to%20update%20the%20Jenkinsfile) says that error will happen if Jenkinsfile isn't updated to remove Java 8 configurations. You've done that, and I don't know what else needs to change.

I know, and it is because I don't have right access on this repo'.

In your link, there is a section talking about this problem =>
```
Note that changes to a plugin’s Jenkinsfile require write access to take effect in pull request builds. If you submit a pull request to a repository to which you do not have write access, any Jenkinsfile changes will be ignored with this message:
```
`Jenkinsfile has been modified in an untrusted revision`

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Google Chat Notification Jenkins Plugin to send build status to [Google Chat](https://chat.google.com)
 
 This Jenkins plugin allows you to send Google Chat notification as a post build action or as a pipeline script.
-This plugin is supported for Jenkins version **2.60.3 or higher**.
+This plugin is supported for Jenkins version **2.387.1 or higher**.
```
Comment (Ryudo302) [inline README.md:6]: Since #31 reported that the problem starts in Jenkins version 2.379, couldn't we start from this version?

File: pom.xml [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
+        <version>4.55</version>
         <relativePath />
     </parent>
     <artifactId>google-chat-notification</artifactId>
     <version>1.6-SNAPSHOT</version>
     <packaging>hpi</packaging>
     <properties>
-        <jenkins.version>2.60.3</jenkins.version>
-        <java.version>1.8</java.version>
+        <jenkins.version>2.387.1</jenkins.version>
```
Comment (Ryudo302) [inline pom.xml:14]: See last comment about the Jenkins version.

**Comment by Ryudo302** [comment]
> > The build is failing with this error: `[2023-03-29T08:44:18.252Z] Caused by: java.lang.UnsupportedClassVersionError: org/jenkinsci/maven/plugins/hpi/ValidateMojo has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0`
> > Build log shows it is building with Java 8: `[2023-03-29T08:43:48.131Z] Java version: 1.8.0_362, vendor: Temurin, runtime: C:\tools\jdk-8\jre`
> > [This blog](https://www.jenkins.io/blog/2022/12/14/require-java-11/#:~:text=If%20you%20neglect%20to%20update%20the%20Jenkinsfile) says that error will happen if Jenkinsfile isn't updated to remove Java 8 configurations. You've done that, and I don't know what else needs to change.
> 
> I know, and it is because I don't have right access on this repo'.
> 
> In your link, there is a section talking about this problem =>
> 
> ```
> Note that changes to a plugin’s Jenkinsfile require write access to take effect in pull request builds. If you submit a pull request to a repository to which you do not have write access, any Jenkinsfile changes will be ignored with this message:
> Jenkinsfile has been modified in an untrusted revision
> 
> Pending JENKINS-46795, the pull request will need to be refiled by someone with write access to ensure the Jenkinsfile changes take effect.
> ```

@regazzoj, I tried to fix the problem with another commit in the Jenkinsfile, but apparently it had no effect. Should I close this PR and open another from your fork?

File: README.md (hunk 1 above)
Comment (regazzoj) [reply README.md:6]: Sounds good to me ! Thanks (:

**Comment by regazzoj** [comment]
> > > The build is failing with this error: `[2023-03-29T08:44:18.252Z] Caused by: java.lang.UnsupportedClassVersionError: org/jenkinsci/maven/plugins/hpi/ValidateMojo has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0`
> > > Build log shows it is building with Java 8: `[2023-03-29T08:43:48.131Z] Java version: 1.8.0_362, vendor: Temurin, runtime: C:\tools\jdk-8\jre`
> > > [This blog](https://www.jenkins.io/blog/2022/12/14/require-java-11/#:~:text=If%20you%20neglect%20to%20update%20the%20Jenkinsfile) says that error will happen if Jenkinsfile isn't updated to remove Java 8 configurations. You've done that, and I don't know what else needs to change.
> > 
> > 
> > I know, and it is because I don't have right access on this repo'.
> > In your link, there is a section talking about this problem =>
> > ```
> > Note that changes to a plugin’s Jenkinsfile require write access to take effect in pull request builds. If you submit a pull request to a repository to which you do not have write access, any Jenkinsfile changes will be ignored with this message:
> > Jenkinsfile has been modified in an untrusted revision
> > 
> > Pending JENKINS-46795, the pull request will need to be refiled by someone with write access to ensure the Jenkinsfile changes take effect.
> > ```
> 
> @regazzoj, I tried to fix the problem with another commit in the Jenkinsfile, but apparently it had no effect. Should I close this PR and open another from your fork?

I don't know... It might be a good try to give !

**Comment by Ryudo302** [comment]
@regazzoj @gpflaum 
I think that I finally solved the problem :)
I had to change the Jenkinsfile myself at a new branch created from master (called _master_jdk11_). Then, I merged @regazzoj's branch into that branch. Finally, I merged the _master_jdk11_ into master.

**Pull request closed without merging**